common/src/main/java/org/apache/celeborn/common/protocol/StorageInfo.java

```diff
@@ -23,10 +23,13 @@
 public class StorageInfo implements Serializable {
   public enum Type {
     MEMORY(0),
+    @Deprecated
     HDD(1),
+    @Deprecated
```

Comment:
Emm... why deprecate them?

Reply:
Because we distribute slots according to device fetch time and write time instead of device type. So the idea is to treat HDD and SSD as same storage tier.
Currently available tiers are Local and HDFS, MEMORY and OSS will be available soon.

Reply:
To keep compatible with old clients, HDD and SSD cannot be deleted or renamed. So I just deprecate them.

Reply:
From the user aspect, I'd like to keep HDD and SSD, because it's already used and easy to be accepted by users, such breaking change does not bring many benefits.

Also, MEMORY, LOCAL, OSS, HDFS is not ideal too, OSS/HDFS should be clustered into REMOTE, right?

Reply:
Sounds good to me. I'll keep the `storageInfo.type` and use the mask to indicate storage tiers.